### PR TITLE
Fixes #34978: Do not use Apipie DSL cache

### PR DESCRIPTION
**Pull request by ehelms**
This makes a change in production away from using an Apipie DSL cache,
and instead relying on HTML docs and JSON being generated on demand.
Testing has found that they are roughly equivalent in load times.
By switching to on-demand, the Apipie DSL documentation can accurately
reflect the state of the running machine and the development and
production environments get closer in alignment. Additionally, this
speeds up installations runs by shaving off around 2 minutes every time
the cache must be generated.

### CODE REVIEW

**Comment by ehelms** [comment]
@ofedoren @evgeni This follows on from the previous Apipie work to drop caching across application code and installer. From my testing, this rake task takes just over 2 minutes any time it has to run with a Katello based setup. The docs page rendered in about 1 second for me which feels acceptable given the improvement to installation time which happens many, many times over through CI and user usage.

**Comment by evgeni** [comment]
Is there a way we could hit the apipie and apipiedsl endpoints in tests, to ensure they actually work?

Sorta unrelated to this change, but would prevent breakage if we for example update something like Rails? ;)

**Comment by ehelms** [comment]
> Is there a way we could hit the apipie and apipiedsl endpoints in tests, to ensure they actually work?

Perhaps hitting the endpoints in the integration tests? Would a 200 on the endpoints be sufficient?

**Comment by evgeni** [comment]
integration = bats? That's IMHO too late.

**Comment by ehelms** [comment]
> integration = bats? That's IMHO too late.

integration = integration? https://github.com/theforeman/foreman/tree/develop/test/integration is what I was pondering

**Comment by evgeni** [comment]
> > integration = bats? That's IMHO too late.
> 
> integration = integration? https://github.com/theforeman/foreman/tree/develop/test/integration is what I was pondering

Ah, right!

Yes, that would work I think

**Comment by ehelms** [comment]
Added two simple tests to fetch the URLs and check for success

**Comment by evgeni** [comment]
Interestingly, disabling caching "fixes" the ActionView incompatibility in ApiepieDSL 2.4.0 (https://github.com/ofedoren/apipie-dsl/commit/c9ce6de7c140729b317fa1715eb413af9b9986ba), as that code isn't called outside the cache generation?

File: config/initializers/apipie.rb [hunk 1]
```diff
@@ -5,7 +5,6 @@
   config.app_name = 'Foreman'
   config.app_info = 'The Foreman is aimed to be a single address for all machines life cycle management.'
   config.doc_base_url = '/templates_doc'
-  config.markup = ApipieDSL::Markup::Markdown.new if Rails.env.development? && defined? Maruku
```
Comment (evgeni) [inline config/initializers/apipie.rb:8]: why is this required?

Comment (ofedoren) [reply config/initializers/apipie.rb:8]: AFAIK we don't ship `maruku` gem, so there is no point to use `Mardown` markup for descriptions in dev mode since they will be broken in prod.

**Review by ofedoren — APPROVED** [review]
LGTM :)

**Review by evgeni — APPROVED** [review]
